tests/integ/sagemaker/lineage/test_association.py

```diff
@@ -54,8 +53,29 @@ def test_list(association_objs, sagemaker_session):
     assert association_keys_listed
 
 
-@pytest.mark.skip(reason="Not in CMH yet")
 def test_set_tag(association_obj, sagemaker_session):
     tag = {"Key": "foo", "Value": "bar"}
     association_obj.set_tag(tag)
-    assert association_obj.get_tag() == tag
+
+    while True:
+        actual_tags = sagemaker_session.sagemaker_client.list_tags(
+            ResourceArn=association_obj.source_arn
+        )["Tags"]
+        if actual_tags:
+            break
+    assert len(actual_tags) == 1
+    assert actual_tags[0] == tag
+
+
+def test_tags(association_obj, sagemaker_session):
+    tags = [{"Key": "foo1", "Value": "bar1"}]
+    association_obj.set_tags(tags)
+
+    while True:
+        actual_tags = sagemaker_session.sagemaker_client.list_tags(
+            ResourceArn=association_obj.source_arn
+        )["Tags"]
+        if actual_tags:
+            break
+    assert len(actual_tags) == 1
+    assert actual_tags == tags
```

Comment:
these while true blocks at the very minimum need a time.sleep, ideally though we would follow best practices using  retry with backoff

Reply:
Okay. Does a sleep time of 1 second sound good?

Also when I noticed the other tests for actions, artifacts or contexts, None of them had sleep in while true blocks and as such followed similar pattern.

Reply:
1 sec is good for now. Those other instances of `while True:` without sleep need to be updated.

Reply:
Added, Thank you